packages/types/src/type/SignaturePayload.ts

```diff
@@ -44,10 +49,10 @@ function sign (signerPair: IKeyringPair, u8a: Uint8Array): Uint8Array {
 export default class SignaturePayload extends Struct {
   protected _signature?: Uint8Array;
 
-  constructor (value?: SignaturePayloadValue | Uint8Array) {
+  constructor (value?: SignaturePayloadValue | Uint8Array, options: ConstructorOptions = {}) {
     super({
       nonce: Nonce,
-      method: Method,
+      method: (isU8a(value) && options.meta) ? Method.withMeta(options.meta) : Method,
```

Comment:
right, I agree it's a bit confusing
I guess I was trying to make explicit that if we pass in a `value: SignaturePayloadValue`, then `options.meta` will not be taken into account, because the `Method` is already instanciated inside `value` (as `value.method`)

Reply:
Ah I get it now. So maybe
```
value instanceOf SignaturePayloadValue ? Method : Method.withMeta(options.meta)
```

the problem is that `AnyU8a = Uint8Array | Array<number> | string`

Reply:
`SignaturePayloadValue` is a `type`, so can't use `instanceof`: ` 'SignaturePayloadValue' only refers to a type, but is being used as a value here.ts(2693)`

> the problem is that AnyU8a = Uint8Array | Array<number> | string

what do you mean?

Reply:
Ah yeah, true.

> what do you mean?

Ah sorry, I was actually looking at the constructor of Extrinsic. There the isU8a seems a bit weirder, we can (and we do) construct extrinsics by hex. `new Extrinsic('0x1234', { meta })` will never use meta

Reply:
What do you think of removing the isU8a part here...

(to be continued, see below)